publiq-ui/pub-link.vue

```diff
@@ -21,7 +21,7 @@
 </script>
 
 <style lang="scss" scoped>
-  a {
+  .link {
```

Comment:
what if I pass something else than a 'a'-tag. Like an Icon or button, it will get the color udb-blue then. Why doesn't the a-tag selector work?

Reply:
maybe a.link as selector?

Reply:
Ah yes, good spot!

Two reasons we shouldn't use `a` on its own:
- It seems like we still have some issues with bootstrap being included _after_ our own styles. But only in Storybook. We should fix that on its own, but it also indicates that our components' selectors are too generic and can easily be overwritten from the outside. Using a class makes our selectors more specific and harder to accidentally overwrite.
- Vue itself recommends not to use html tags as selectors when using scoped for performance reasons: https://vue-loader.vuejs.org/guide/scoped-css.html#also-keep-in-mind

> Scoped styles do not eliminate the need for classes. Due to the way browsers render various CSS selectors, p { color: red } will be many times slower when scoped (i.e. when combined with an attribute selector). If you use classes or ids instead, such as in .example { color: red }, then you virtually eliminate that performance hit.

I'll see if I can apply the class only if it's an `a`, or go with your suggestion 👍

Reply:
Here's a screenshot of reboot overwriting our own styles in Storybook (the stylesheets file names are only visible in dev mode btw, on Chromatic it's all inline but also in this order)

![Screenshot 2020-09-22 at 14 36 09](https://user-images.githubusercontent.com/959026/93882895-1ba71f80-fce1-11ea-91c8-6eb826234c8a.png)

Reply:
Ah, I just realized it's not the order in which the CSS is included or loaded, but the precedence of the selectors. Reboot's `a:not([href]):not([class])` is simply more specific than our own `a` selector which is why it takes precedence.

Reply:
Adding `href="#"` to the template of the story fixes it. (It was in the example docblock, but not the template of the actual story.) But then on hover I get a similar problem:

![Screenshot 2020-09-22 at 14 42 26](https://user-images.githubusercontent.com/959026/93883505-f2d35a00-fce1-11ea-9221-0a080bed7552.png)

Reply:
@simon-debruijn I've implemented your suggestion + it now only adds the class if the link is an `a` (to avoid any conflicts with another `.link` class outside the component) + added an `href` to the story template.